homeassistant/components/samsungtv/media_player.py

```diff
@@ -93,13 +93,21 @@ def __init__(self, config_entry, on_script):
         # Mark the end of a shutdown command (need to wait 15 seconds before
         # sending the next command to avoid turning the TV back ON).
         self._end_of_power_off = None
-        # Initialize bridge
-        self._bridge = SamsungTVBridge.get_bridge(
-            config_entry.data[CONF_METHOD],
-            config_entry.data[CONF_HOST],
-            config_entry.data[CONF_PORT],
-            config_entry.data.get(CONF_TOKEN),
-        )
+
+        data = config_entry.data.copy()
+        while True:
+            # Initialize bridge
+            self._bridge = SamsungTVBridge.get_bridge(
+                data[CONF_METHOD],
+                data[CONF_HOST],
+                data[CONF_PORT],
+                data.get(CONF_TOKEN),
+            )
+            if self._bridge.port is None and self._bridge.default_port is not None:
+                data[CONF_PORT] = self._bridge.default_port
+                hass.config_entries.async_update_entry(config_entry, data=data)
```

Comment:
We don't want side effects in init method. Probably move the whole bridge creation, including config entry update, outside the entity in `async_setup_entry` and pass the bridge to the entity instead.

Reply:
If we adjust per above, I think this is ok.

Reply:
agree

Reply:
@MartinHjelmare I did what you asked but some tests in test_media_player started failing :(
Any clue?

Reply:
Silly error :( should be fixed now.